File: 0162-find-peak-element/0162-find-peak-element.java
class Solution {
    public int findPeakElement(int[] nums) {
        int len = nums.length;

        for (int i = 0; i < len; i++) {
            if ((i == 0 || nums[i] > nums[i - 1]) && (i == len - 1 || nums[i] > nums[i + 1])) {
                return i; // Peak element found
            }
        }
        
        return -1; // Edge case (should never reach here)
    }
}

File: 0162-find-peak-element/0162-find-peak-element.py
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        left, right = 0, len(nums) - 1

        while left < right:
            mid = left + (right - left) // 2

            if nums[mid] > nums[mid + 1]:  
                right = mid  # Peak is on the left side
            else:
                left = mid + 1  # Peak is on the right side
        
        return left  # The peak index